src/main/kotlin/coffeebot/database/WagerDao.kt

```diff
@@ -0,0 +1,125 @@
+package coffeebot.database
+
+import org.jetbrains.exposed.sql.ResultRow
+import org.jetbrains.exposed.sql.Table
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.insertAndGetId
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.jetbrains.exposed.sql.update
+
+// Represents result of a transaction.
+sealed class Result {
+    object Success : Result()
+    object Failure : Result()
+}
+
+/**
+ * Propose a CoffeeWager.
+ * @param person: Person proposing wager.
+ * @param coffeesFor: Number of coffees proposer wishes to wager.
+ * @param coffeesAgainst: Number of coffees proposer wishes opponent to wager.
+ * @param termsOfWager: What the dang thing is about!
+ */
+fun proposeWager(person: String, coffeesFor: Int, coffeesAgainst: Int, termsOfWager: String): Int {
+    return transaction {
+        CoffeeWager.insertAndGetId {
+            it[person1] = person
+            it[coffees1] = coffeesFor
+            it[coffees2] = coffeesAgainst
+            it[terms] = termsOfWager
+            it[state] = WagerState.Proposed
+        }
+    }.value
+}
+
+/**
+ * Cancel a proposed wager.
+ * @param person: Person cancelling wager.
+ * @param id: Id of bet to cancel.
+ */
+fun cancelWager(person: String, id: Int): Result {
+    return transaction {
+        val before = getCanceledWagers().size
+
+        CoffeeWager.update ({ (CoffeeWager.id eq id)
+                .and( CoffeeWager.person1 eq person)
+                .and(CoffeeWager.state eq WagerState.Proposed)}) {
+            it[state] = WagerState.Canceled
+        }
+
+        val after = getCanceledWagers().size
```

Comment:
This works but feels a little contrived... Can't you just check that the wager is in a valid state, and then unconditionally update it? I think it'd work since you're doing this in a transaction.

It would be nice if `.update` just told you how many rows were touched, but I assume it doesn't do that (I can't find the doc for it for some reason).

Same comment below.